### DISCUSSION TITLE
Asynchronous call traceid value transfer

### DISCUSSION THREAD

**Discussion by wealthtears** [apache/skywalking#7806] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/35245652/134805049-e9f8fb9a-3717-4982-8de2-5ff33d611c01.png)
![image](https://user-images.githubusercontent.com/35245652/134805061-b9e91974-a08c-4373-86e5-20583e5702ee.png)

This code is a simple example written for questioning. The original system is complex. It is hoped that when sending asynchronous requests, you can set the request header and change the traceid value to achieve link consistency.

Is there any good solution,pleases!

**Comment by wu-sheng** [comment]
Do you read SkyWalking protocols? I feel you don't use SkyWalking. Because this is not how we provide to trace HTTP. We have supported many async frameworks of RPCs and schedulers.

**Reply by wu-sheng** [reply]
Tracing an async process is not forwarding trace id only. Your trace breaks still

**Reply by wealthtears** [reply]
Thank you for your answer.
I use Seda high concurrency framework. This framework model is event driven and segmented processing. Under this framework model, using skywalking multithreading still can not realize link tracking. In the past, when integrating zkpin links, I artificially modified the header attribute traceid when sending requests for the next event.

**Reply by wealthtears** [reply]
Is it because the requests are consumed from the thread pool, but skywalking doesn't know the specific thread, so it can't use the plug-in you provide

**Reply by wu-sheng** [reply]
I think plugin has supported the across thread case, which is very common. React, async framework, etc. are all using it. SkyWalking doesn't need to know the thread at all. Please read the APIs of plugin.
What you showed on the codes is not official way, so, we wouldn't follow your path.

**Comment by wealthtears** [comment, marked as answer]
The plug-in provided by skywalking does support asynchronous multithreading, but my actual situation is thread pool / event routing / multithreading. Now, I have solved my problem with plug-in development. Thank you!
